src/data/move-attrs/stat-stage-change-all-opps-attr.ts

```diff
@@ -0,0 +1,108 @@
+import { type BattleStat } from "#enums/stat";
+import { type Pokemon } from "#app/field/pokemon";
+import { globalScene } from "#app/global-scene";
+import { StatStageChangePhase } from "#app/phases/stat-stage-change-phase";
+import type { Move } from "#app/data/move";
+import { type MoveEffectAttrOptions, MoveEffectAttr } from "#app/data/move-attrs/move-effect-attr";
+import type { MoveConditionFunc } from "../move-conditions";
+
+/**
+ * Set of optional parameters that may be applied to stat stage changing effects
+ * @extends MoveEffectAttrOptions
+ * @see {@linkcode StatStageChangeAttr}
+ */
+interface StatStageChangeAllOppsAttrOptions extends MoveEffectAttrOptions {
+  /** If defined, needs to be met in order for the stat change to apply */
+  condition?: MoveConditionFunc;
+  /** `true` to display a message */
+  showMessage?: boolean;
+}
+
+/**
+ * Attribute used for G-Max moves that change the stats of all opponents
+ *
+ * +-------------+----------+--------+
+ * | G-Max Move  |   Stat   | Change |
+ * +-------------+----------+--------+
+ * | Foam Burst  | Speed    |     -2 |
+ * | Tartness    | Evasion  |     -1 |
+ * +-------------+----------+--------+
+ *
```

Comment:
```suggestion
 * ```
 * +-------------+----------+--------+
 * | G-Max Move  |   Stat   | Change |
 * +-------------+----------+--------+
 * | Foam Burst  | Speed    |     -2 |
 * | Tartness    | Evasion  |     -1 |
 * +-------------+----------+--------+
 * ```
```